### ISSUE TITLE
fine-tune for different categories

### ISSUE THREAD

**Issue by suhyung** [weiliu89/caffe#162]
If I hope to get 5 categories detection, can't I use fine-tune based on PASCAL-VOC which has 21 categories? My own DB has just 5 categories and they are not overlapped with PACAL-VOC's.

I tried to change 'num_classes' value on ssd_pascal.py,  and "Cannot copy param 0 weights from layer 'conv4_3_norm_mbox_conf'; shape mismatch" message shown.

**Comment by yeongrok** [comment]
Which pretrained model did you use?  If you used "VGG_ILSVRC_16_layers_fc_reduced.caffemodel", I guess it should be ok,  but if you used fine-tuned model by PASCAL data,  you will get error because of the difference of the num of categories.
Simple solution is to change layer's name in your prototxt. 
eg. conv4_3_norm_mbox_loc --> conv4_3_norm_mbox_loc_xx, 
     conv4_3_norm_mbox_conf --> conv4_3_norm_mbox_conf_xx,
     fc7...

**Comment by weiliu89** [comment]
@yeongrok is almost right. You could also try to keep the name for the location prediction branch because it is shared for all classes, and only change the name and fine-tune the classification path. Of course, you may want to try smaller learning rate.

**Comment by suhyung** [comment]
@yeongrok Thanks. You are correct. As you said, VGG_ILSVRC_16_layers_fc_reduced.caffemodel works well as fine-tune base model. but I don't understand "change layer's name" . Does that mean that remove the conv4_3_norm_mbox_conf from mbox_conf's buttom? why the name should be changed?

@weiliu89 Thanks for response. could you explain your mention "only change the name and fine-tune the classification path"?  what name should be changed? and what is classification path? if classification path means labelmap_voc.prototxt, I already did it.

**Comment by yeongrok** [comment]
@suhyung As you may know, prototxt defines network structure and each layer in the network has name. See prototxt file and change the layer's name
Classification path means layers in a network. Those are in charge of classification in the network.

**Comment by Anhaoxu** [comment]
hello, @suhyung @yeongrok @weiliu89 
I got the same mistake, if I use the VGG_ILSVRC_16_layers_fc_reduced.caffemodel, it worked well, but I use the VGG_VOC0712_SSD_300x300_iter_120000.caffemodel, when running the ssd_pascal_video.py, the same mistake occurred,then I changed the two layers name: conv4_3_norm_mbox_loc and conv4_3_norm_mbox_conf, but the result shows that there are too many bbox occurred and they are wrong. 
Finally, I'd like to look for the bbox coordinates in the py code, is there any detailed method to do this and solve the above problem?
Thank you!

**Comment by weiliu89** [comment]
You could provide conf_postfix in the CreateMultiBoxHead function in your python script
